crates/vim/src/helix.rs

```diff
@@ -226,6 +227,33 @@ impl Vim {
             _ => self.helix_move_and_collapse(motion, times, cx),
         }
     }
+
+    pub fn helix_delete(&mut self, _: &HelixDelete, cx: &mut ViewContext<Self>) {
+        self.store_visual_marks(cx);
+        self.update_editor(cx, |vim, editor, cx| {
+            // Fixup selections so they have helix's semantics.
+            // Specifically:
+            //  - Make sure that each cursor acts as a 1 character wide selection
+            editor.transact(cx, |editor, cx| {
+                editor.change_selections(Some(Autoscroll::fit()), cx, |s| {
+                    s.move_with(|map, selection| {
+                        if selection.is_empty() && !selection.reversed {
+                            *selection.end.column_mut() += 1;
```

Comment:
@leroycep it is not safe to +1 to a column without clipping it, as you may put the cursor in the middle of a multlibyte utf8 sequence.

It'd be better to do `selection.end = movement::right(selection.end)`

Not sure how soon we want to support `.` for helix,

Reply:
updated to use `movement::right`

Reply:
I think that means we don't need the second condition as right at the end of the line will move the start of the next?

Reply:
Yep, that seems to be the case. The tests I added still pass with it removed.

Does the `!selection.reversed` check need to be there? I'm not sure if it's possible for a selection to be reversed and empty.

Reply:
It's not needed, reversed is only if head < tail.